lib/Master.php

```diff
@@ -123,8 +126,13 @@ public function handleLoginRequest($param) {
 			return;
 		}
 
-		if (empty($location)) {
-			$location = $this->queryLookupServer($uid);
+		// first ask the lookup server if we already know the user
+		$location = $this->queryLookupServer($uid);
+		// if not we fall-back to a initial user deployment method, if configured
+		if (empty($location) && !empty($userDiscoveryModule) && class_exists($userDiscoveryModule)) {
```

Comment:
IMO we should log that the `class_exists` failed - otherwise it's super hard to spot typos in the configuration because you don't know if it is a bug or just a typo, because there is exactly 0 log about this.

I would put it in as a warning.